test/vagrant-service-manager/binary_handlers/adb_docker_binary_handler_test.rb

```diff
@@ -0,0 +1,100 @@
+require 'test_helper'
+require 'vagrant/util/downloader'
+
+# Tests through ADBDockerBinaryHandler to BinaryHandler
+module VagrantPlugins
+  module ServiceManager
+    describe ADBDockerBinaryHandler do
+      before do
+        @machine = fake_machine
+        @options = { type: :docker, '--cli-version' => '1.11.0' }
+
+        # set test path
+        @plugin_test_path = "#{@machine.env.data_dir}/service-manager/test"
+        ServiceManager.temp_dir = "#{@plugin_test_path}/temp"
+        ServiceManager.bin_dir = "#{@plugin_test_path}/bin"
+
+        @handler = ADBDockerBinaryHandler.new(@machine, @machine.env, @options)
+      end
+
+      after do
+        FileUtils.rmtree(@plugin_test_path) if File.directory? @plugin_test_path
+      end
+
+      it 'should set defaults values properly' do
+        @handler.instance_variable_get('@machine').must_equal @machine
+        @handler.url.must_equal ''
+        @handler.binary_exists.must_equal true
+        @handler.skip_download.must_equal false
+        @handler.archive_file_path.must_equal ''
+        @handler.type.must_equal @options[:type]
+        @handler.version.must_equal @options['--cli-version']
+        @handler.path.must_equal "#{ServiceManager.bin_dir}/docker/1.11.0/docker"
+        expected_temp_bin_dir = "#{ServiceManager.temp_dir}/docker"
+        @handler.instance_variable_get('@temp_bin_dir').must_equal expected_temp_bin_dir
+      end
+
+      it 'should build download url' do
+        expected_url = 'https://get.docker.com/builds/Linux/x86_64/docker-1.11.0.tgz'
```

Comment:
This is OS specific. Does not pass on OS X. You will check the OS type when building the expectation.